gold_digger/data_providers/_provider.py

```
@@ -10,55 +10,84 @@
+        """
+        raise NotImplementedError
 
     @abstractmethod
-    def get_all_by_date(self, date_of_exchange, currencies):
-        pass
+    def get_all_by_date(self, date_of_exchange, currencies, logger):
+        """
+        :type date_of_exchange: datetime.date
+        :type currencies: [str]
+        :type logger: gold_digger.utils.context_logger.ContextLogger
+        :rtype: {str: decimal.Decimal | None}
+        """
+        raise NotImplementedError
 
     @abstractmethod
-    def get_historical(self, origin_date, currencies):
-        pass
+    def get_historical(self, origin_date, currencies, logger):
+        """
+        :type origin_date: datetime.date
+        :type currencies: [str]
+        :type logger: gold_digger.utils.context_logger.ContextLogger
+        :rtype: {datetime.Datetime: {str: decimal.Decimal | None}}
+        """
+        raise NotImplementedError
 
-    def _get(self, url, params=None):
+    def _get(self, url, params=None, *, logger):
+        """
+        :type url: str
+        :type params: {str: str}
+        :type logger: gold_digger.utils.context_logger.ContextLogger
+        :rtype: requests.Response | None
+        """
         try:
             response = requests.get(url, params=params, timeout=self.DEFAULT_REQUEST_TIMEOUT)
             if response.status_code == 200:
                 return response
             else:
-                self.logger.error("%s - status code: %s, URL: %s, Params: %s", self, response.status_code, url, params)
+                logger.error("%s - status code: %s, URL: %s, Params: %s", self, response.status_code, url, params)
         except requests.exceptions.RequestException as e:
-            self.logger.error("%s - exception: %s, URL: %s, Params: %s", self, e, url, params)
+            logger.error("%s - exception: %s, URL: %s, Params: %s", self, e, url, params)
 
-    def _to_decimal(self, value, currency=None):
+    def _to_decimal(self, value, currency=None, *, logger):
```

@Joozty I don't know if is it ok to pass a logger instance to function `_to_dacimal`. I expect that this function needs a value to convert but why I should pass a logger there?

---

https://github.com/business-factory/gold-digger/pull/60/commits/d28d68b6d0ead7a57b0067bafb073264e1d010ff

---

So, you removed the log and you just silently catch the exception. What if a provider will change something in the API and the conversion will start failing? We will not know what's going on. Luckily, this hasn't happened for a long time, but the log was there on purpose. If you don't like passing logger to the function, I suggest to raise the exception and catch it within provider, but I guess it's better to report possible errors.